controllers/users.js

```diff
@@ -0,0 +1,174 @@
+const mongoose = require('mongoose')
+const Courses = require('../models/courses')
+const bcrypt = require('bcryptjs')
+const { Users, Roles } = require('../models/users')
+
+const userController = {};
+
+/**
+ * @param {Object} req 
+ * @param {Object} res 
+ * @param {Function} next 
+ * @description Fetches all users' data 
+ */
+userController.getAll = async (req, res) => {
+    try {
+        const allUsera = await Users.find()
```

Comment:
this is a wrong method of filtering the data, it should be handled in the query itself
`const allUsers = await Users.find({ is_deleted: false });`